QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryController.java

```diff
@@ -50,55 +72,416 @@ public void init() {
+				.onItem().transform(map -> {
+					if (map.isEmpty()) {
+						return RestResponse.notFound();
+					} else {
+						return HttpUtils.generateEntityResult(contextHeader, context, acceptHeader, map, null, null,
+								null);
+					}
+				});
+
 	}
 
 	@Path("/attributes")
 	@GET
 	public Uni<RestResponse<Object>> getAllAttributes(HttpServerRequest request,
-			@QueryParam(value = "details") boolean details) {
+			@QueryParam(value = "details") boolean details, @QueryParam(value = "localOnly") boolean localOnly) {
+
+		int acceptHeader = HttpUtils.parseAcceptHeader(request.headers().getAll(HttpHeaders.ACCEPT));
+		if (acceptHeader == -1) {
+			return HttpUtils.getInvalidHeader();
+		}
+		if (!details) {
+			return queryService.getAttribs(HttpUtils.getTenant(request), localOnly).onItem().transform(map -> {
+				List<Object> contextHeader = HttpUtils.getAtContext(request);
+				return HttpUtils.generateEntityResult(contextHeader,
+						JsonLdProcessor.getCoreContextClone().parse(contextHeader, true), acceptHeader, map, null, null,
+						null);
+			});
+		} else {
+			return queryService.getAttribsWithDetails(HttpUtils.getTenant(request), localOnly).onItem()
+					.transform(list -> {
+						List<Object> contextHeader = HttpUtils.getAtContext(request);
+						return HttpUtils.generateEntityResult(contextHeader,
+								JsonLdProcessor.getCoreContextClone().parse(contextHeader, true), acceptHeader, list,
+								null, null, null);
+					});
+		}
 
-		return QueryControllerFunctions.getAllAttributes(queryService, request, details, false, defaultLimit, maxLimit);
 	}
 
 	@Path("/attributes/{attribute}")
 	@GET
 	public Uni<RestResponse<Object>> getAttribute(HttpServerRequest request, @PathParam("attribute") String attribute,
-			@QueryParam(value = "details") boolean details) {
-		return QueryControllerFunctions.getAttribute(queryService, request, attribute, details, false, defaultLimit,
-				maxLimit);
+			@QueryParam(value = "details") boolean details, @QueryParam(value = "localOnly") boolean localOnly) {
+		int acceptHeader = HttpUtils.parseAcceptHeader(request.headers().getAll(HttpHeaders.ACCEPT));
+		if (acceptHeader == -1) {
+			return HttpUtils.getInvalidHeader();
+		}
+		Context context;
+		List<Object> contextHeader = HttpUtils.getAtContext(request);
+		if (!contextHeader.isEmpty()) {
+			context = JsonLdProcessor.getCoreContextClone().parse(contextHeader, null, true);
+		} else {
+			context = JsonLdProcessor.getCoreContextClone();
+		}
+		attribute = context.expandIri(attribute, false, true, null, null);
+		return queryService.getAttrib(HttpUtils.getTenant(request), attribute, localOnly).onItem().transform(map -> {
+			if (map.isEmpty()) {
+				return RestResponse.notFound();
+			} else {
+				return HttpUtils.generateEntityResult(contextHeader, context, acceptHeader, map, null, null, null);
+			}
+		});
+
+	}
+
+	@Path("/entityoperations/query")
```

Comment:
It should be "/entityOperations/query" instead of "/entityoperations/query" means "o" sould caps.

Reply:
done